### PR TITLE
Collective2 module modernization and finalization

### PR DESCRIPTION
**Pull request by francis2**
Removed platformId support from C2.
Modernized C2 API calls to APIv4.
Added C2 rate limiters (RateGate).
Added response logging so users can easily debug their code. Added a few symbols for the demo.

#### Description
Collective2 Module and a DemonstrationAlgorithm

#### Related Issue
I am not aware of any related open issues

#### Motivation and Context
Modernize and finalize module

#### Requires Documentation Change
None

#### How Has This Been Tested?
Collective2SignalExportDemonstrationAlgorithm contains a list of symbols that produce expected results.

#### Types of changes

- [X ] Bug fix (non-breaking change which fixes an issue)
- [X ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [ X] My code follows the code style of this project.
- [ X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ X] I have added tests to cover my changes. 
- [X ] All new and existing tests passed.
- [ X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Algorithm/Framework/Portfolio/SignalExports/Collective2SignalExport.cs [hunk 1]
```diff
@@ -224,49 +230,163 @@ protected string CreateMessage(List<Collective2Position> positions)
...
             }
+            else if (responseObject.Results.Count > 0)
+            {
+                _algorithm.Debug($"Collective2: NewSignals={string.Join(',', responseObject.Results[0].NewSignals)} | CanceledSignals={string.Join(',', responseObject.Results[0].CanceledSignals)}");
+            }
+            
+            return true;
+        }
+
+        static string PrintErrors(List<ResponseError> errors)
```
Comment (Martin-Molinero) [inline Common/Algorithm/Framework/Portfolio/SignalExports/Collective2SignalExport.cs:259]: Please add the missing `private` access modifier to match existing code style👍

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Hey @francis2! Thank you for the PR, leaving a couple of requests 👍

File: Algorithm.CSharp/Collective2PortfolioSignalExportDemonstrationAlgorithm.cs [hunk 2]
```diff
@@ -32,20 +34,16 @@ namespace QuantConnect.Algorithm.CSharp
     public class Collective2PortfolioSignalExportDemonstrationAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
```
[the remark this answers is not in the document]
Comment (Martin-Molinero) [reply Algorithm.CSharp/Collective2PortfolioSignalExportDemonstrationAlgorithm.cs:34]: It seems another unit test needs an update too 
image

File: Common/Algorithm/Framework/Portfolio/SignalExports/Collective2SignalExport.cs [hunk 3]
```diff
@@ -224,49 +230,163 @@ protected string CreateMessage(List<Collective2Position> positions)
         private bool SendPositions(string message)
         {
             using var httpMessage = new StringContent(message, Encoding.UTF8, "application/json");
+
+            //Add the QuantConnect app header
+            httpMessage.Headers.Add("X-AppId", "OPA1N90E71");
+
+            //Add the Authorization header
+            HttpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", _apiKey);
```
[the remark this answers is not in the document]
Comment (francis2) [reply Common/Algorithm/Framework/Portfolio/SignalExports/Collective2SignalExport.cs:238]: It needs to be in every message header

**Comment by Martin-Molinero** [comment]
Reviews addressed in https://github.com/QuantConnect/Lean/pull/7448, closing this PR since included

**Pull request closed without merging**